src/shopper.py

```diff
@@ -0,0 +1,318 @@
+        
+        self._screen = screen.Screen(config.general["monitor"])
+        self.config = config
+        self._template_finder = TemplateFinder(self._screen)
+        self._npc_manager = NpcManager(
+            screen=self._screen, template_finder=self._template_finder
+        )
+        self.run_count = 0
+        self.start_time = time.time()
+        self.ias_gloves_seen = 0
+        self.gloves_bought = 0
+        
+        # Claws config
+        self.roi_claw_stats = [0, 0, config.ui_pos["screen_width"] // 2, config.ui_pos["screen_height"] - 100]
+        self.roi_vendor = config.ui_roi["vendor_stash"]
+        self.rx, self.ry, _, _ = self.roi_vendor
+        self.sb_x, self.sb_y = self._screen.convert_screen_to_monitor((180, 77))
+        self.c_x, self.c_y = self._screen.convert_screen_to_monitor((config.ui_pos["center_x"], config.ui_pos["center_y"]))
+        self.claws_evaluated = 0
+        self.claws_bought = 0
+
+    def run(self):
+        Logger.info("Personal Anya Shopper at your service! Hang on, running some errands...")
+        self.reset_shop()
+        self.shop_loop()
+
+    def shop_loop(self):
+        if self.config.general["res"] == "1280_720":
+            asset_folder = "assets/shop_1280_720/gloves/"
+        else:
+            asset_folder = "assets/shop/gloves/"
+        while True:
+            self._npc_manager.open_npc_menu(Npc.ANYA)
+            self._npc_manager.press_npc_btn(Npc.ANYA, "trade")
+            time.sleep(0.1)
+            img = self._screen.grab()
+
+            # 20 IAS gloves have a unique color so we can skip all others
+            ias_glove_found, pos = self._template_finder.search(
+                ref=load_template(asset_folder + "ias_gloves.png", 1.0),
+                inp_img=img,
+                threshold=0.96,
+                roi=self.config.ui_roi["vendor_stash"],
+                normalize_monitor=True,
+            )
+            if ias_glove_found:
+                self.ias_gloves_seen += 1
+                mouse.move(*pos)
+                time.sleep(0.1)
+                img = self._screen.grab()
+                
+                if self.look_for_plus_3_gloves is True:
+                    gg_gloves_found, pos = self._template_finder.search(
+                        ref=load_template(
+                            asset_folder + "gg_gloves.png", 1.0
+                        ),
+                        inp_img=img,
+                        threshold=0.80,
+                        normalize_monitor=True,
+                    )
+                    if gg_gloves_found:
+                        mouse.click(button="right")
+                        send_discord(f"Bought awesome IAS/+3 gloves!", config.general["custom_discord_hook"])
+                        Logger.info("IAS/+3 gloves bought!")
+                        self.gloves_bought += 1
+                        time.sleep(1)
+
+# Enable the following, if you are fine with +2 gloves
```

Comment:
indent

Reply:
Removed, the look_for_plus_2_gloves kinda makes the comment redundant.